src/metabase/models.clj

```diff
@@ -192,3 +198,22 @@
        (when (isa? metabase-models-keyword :metabase/model)
          metabase-models-keyword)))
    (next-method symb)))
+
+(t2/define-after-insert :metabase/model
```

Comment:
wdyt instead of adding this to `:metabase/model,` we create a new model `:search/UpdateIndex` and derive any models that need updating?

Reply:
The main benefits will be that if you read a model file, you'll know that it subscribes to this side-affect

Reply:
I kinda like that, but then there's a risk of forgetting a model.

I guess we could write a test for this though, will give it a shot.